### ISSUE TITLE
Commands don't match username in webhook mode

### ISSUE THREAD

**Issue by Nakochi-crypto** [telegraf/telegraf#1809]
## Context

* Telegraf.js Version: v4.12.0
* Node.js Version: v18.x
* Operating System: N/A

## Minimal Example Code Reproducing the Issue

```ts
bot.command('hi', (ctx) => ctx.reply('hi'));
```

With `/hi`, the command works, but not work with `/hi@My_Bot`. RegExp support might affect.

## Expected Behavior

The bot replies.

## Current Behavior

The bot doesn't reply with commands including the bot name.

## Error Message and Logs (`export DEBUG='telegraf:*'`)

N/A

**Comment by MKRhere** [comment]
My code:

```TS
import { Telegraf } from "telegraf";
const bot = new Telegraf(token);
bot.command("hi", ctx => ctx.reply("hi"));
bot.launch();
```

![image](https://user-images.githubusercontent.com/19621208/223032210-6f14dc1e-047e-4562-add3-0aff8236e4ad.png)

Seems to work fine. To clarify, it will not match `/hi@otherbot`. It'll only match your own bot username.

Are you using webhook mode or some other setup?

**Comment by Nakochi-crypto** [comment]
Yes. I'm using webhook mode on firebase.

My code like this:

```ts
import { Telegraf, Composer } from 'telegraf';

const bot = new Telegraf(TOKEN);

const command = Composer.command('hi', (ctx) => ctx.reply('hi'));
bot.use(command);

export const webhook = bot.webhookCallback('/' + bot.secretPathComponent());
```

**Comment by MKRhere** [comment]
Okay, so this was always broken in webhook mode, even before 4.12.0. If you don't call `bot.launch()`, the initial `getMe` is not called, and the bot doesn't know its own username. What could be the ideal solution to this problem? Could it be solved by documentation?

This line is required:

```TS
bot.botInfo ??= await this.telegram.getMe();
```

Somewhere before your bot starts. We could wrap it in a method such as `await bot.init()` that you have to do before starting the webhook listener. Does this make sense?

**Comment by Nakochi-crypto** [comment]
Unfortunately it doesn't work.

FYI, my code above worked with v4.11.2.

I guess it doesn't matter because `getMe()` is also called in `handleUpdate()`. In proof this code displays the bot username.

```ts
export default async function (
  req: functions.https.Request,
  res: functions.Response
): Promise<void> {
  const webhook = bot.webhookCallback('/' + bot.secretPathComponent());
  await webhook(req, res);
  functions.logger.debug(bot.botInfo?.username);
}
```

**Comment by Nakochi-crypto** [comment]
Did you forget to call `toLowerCase()` with `ctx.me`? It returns case-sensitive bot username.

https://github.com/telegraf/telegraf/blob/361b6664b0be7e231c4ee14fec6f63f1a7a43892/src/composer.ts#L687

The code below shows case-sensitive bot username.

```ts
import { Telegraf } from 'telegraf';

const bot = new Telegraf(TOKEN);
bot.use((ctx) => console.log(ctx.me)); // My_Bot

export const webhook = bot.webhookCallback('/' + bot.secretPathComponent());
```

**Comment by MKRhere** [comment]
> `getMe()` is also called in `handleUpdate()`

You're right, that's true; with that in mind, I tried running your example code as-is, and could not reproduce the issue. Here's my full code:

```TS
import { Telegraf, Composer } from "telegraf";
import Express from "express";

const bot = new Telegraf(token);
bot.command("hi", ctx => ctx.reply("hi"));
const command = Composer.command("hi", ctx => ctx.reply("hi"));
bot.use(command);

const webhook = bot.webhookCallback("/" + bot.secretPathComponent());
const app = Express();
app.use(webhook);
app.listen(10000);

await bot.telegram.setWebhook(domain + "/" + bot.secretPathComponent());
```

If you can create a minimum reproduction like this, please send that, and I'll try. Also, please do try running it on 4.11.2 once again to verify that it was introduced in 4.12.0. I tried running this code in both versions and could not reproduce the issue.

**Comment by MKRhere** [comment]
> Did you forget to call `toLowerCase()` with `ctx.me`?

> It returns case-sensitive bot username.

Aha, I see. Will fix in a patch release.

**Comment by MKRhere** [comment]
Released [`4.12.1`](https://github.com/telegraf/telegraf/releases/tag/v4.12.1), thanks for the excellent issue debugging.

**Comment by Nakochi-crypto** [comment]
No problem. My bot works again. Thanks!
